### PR TITLE
Make compatible with Rust 1.0 beta

### PR DESCRIPTION
**Pull request by frewsxcv**

### CODE REVIEW

**Comment by frewsxcv** [comment]
A couple of the examples are relying on the `tcp` feature, which is unstable, so those will need to be updated

File: .travis.yml [hunk 1]
```diff
@@ -1,4 +1,7 @@
 language: rust
+rust:
+  - nightly
+  - 1.0.0-beta
```
Comment (tomaka) [inline .travis.yml]: Needs to be `1.0.0-beta.4`. You are referring to the first beta version, is different than the current one.

**Comment by frewsxcv** [comment]
@tomaka Unless you want to address the 1.0 incompatibilities sooner, I will attempt to fix this tonight

**Comment by frewsxcv** [comment]
@tomaka how does this stuff look?

**Pull request merged**